regulations/views/mixins.py

```diff
@@ -2,18 +2,23 @@
 
 import six
 from django.conf import settings
+from django.urls import reverse, NoReverseMatch
 
 from regulations.generator import api_reader
 from regulations.generator.toc import fetch_toc
-from regulations.generator.section_url import SectionUrl
 
 
 def build_citation(context):
         citation = []
         if 'part' in context:
             citation.append(context["part"])
+        else:
+            return []
```

Comment:
I messed this up. The function shouldn't return two different types, so this should return an empty string or the following conditionals should be nested inside this if e.g.

```python
        if 'part' in context:
            citation.append(context["part"])
            if 'section' in context:
                citation.append(context["section"])
            elif 'subpart' in context:
                citation.append(context["subpart"])
        return "-".join(citation)
```